test/e2e/network/networking.go

```diff
@@ -98,6 +98,54 @@ var _ = SIGDescribe("Networking", func() {
 		config.GetSelfURLStatusCode(ports.ProxyStatusPort, "/proxyMode", "200")
 	})
 
+	It("should function for client IP based session affinity: http", func() {
+		config := framework.NewNetworkingTestConfig(f)
+		By(fmt.Sprintf("dialing(http) %v --> %v:%v (config.clusterIP)", config.TestContainerPod.Name, config.ClusterIP, framework.ClusterHttpPort))
+		updateSessionAffinity := func(svc *v1.Service) {
+			svc.Spec.SessionAffinity = v1.ServiceAffinityClientIP
+		}
+		_, err := framework.UpdateService(f.ClientSet, config.NodePortService.Namespace, config.NodePortService.Name, updateSessionAffinity)
+		if err != nil {
+			framework.Failf("Failed to update service session affinity, error: %v", err)
+		}
+
+		// debug
+		framework.Logf("Updated service: %v", config.NodePortService)
+
+		// Check if number of endpoints returned are exactly one.
+		eps, err := config.GetDiffEndpointsFromTestContainer("http", config.ClusterIP, framework.ClusterHttpPort, framework.SessionAffinityChecks)
+		if err != nil {
+			framework.Failf("Unexpetced error: %v", err)
+		}
+		if len(eps) > 1 {
+			framework.Failf("Unexpetced endpoints return: %v, expect 1 endpoints", eps)
+		}
+	})
+
+	It("should function for client IP based session affinity: udp", func() {
+		config := framework.NewNetworkingTestConfig(f)
+		By(fmt.Sprintf("dialing(udp) %v --> %v:%v (config.clusterIP)", config.TestContainerPod.Name, config.ClusterIP, framework.ClusterUdpPort))
+		updateSessionAffinity := func(svc *v1.Service) {
+			svc.Spec.SessionAffinity = v1.ServiceAffinityClientIP
+		}
+		_, err := framework.UpdateService(f.ClientSet, config.NodePortService.Namespace, config.NodePortService.Name, updateSessionAffinity)
+		if err != nil {
+			framework.Failf("Failed to update service session affinity, error: %v", err)
+		}
+
+		// debug
+		framework.Logf("Updated service: %v", config.NodePortService)
+
+		// Check if number of endpoints returned are exactly one.
+		eps, err := config.GetDiffEndpointsFromTestContainer("udp", config.ClusterIP, framework.ClusterUdpPort, framework.SessionAffinityChecks)
+		if err != nil {
+			framework.Failf("Unexpetced error: %v", err)
```

Comment:
Would prefer to explain when/why, eg:

"Failed to get endpoints from test container, error: "